### ISSUE TITLE
Can't change default font

### ISSUE THREAD

**Issue by alexsolisc** [eddelbuettel/tint#29]
Hi, 

I couldn't find a way to change the default font for both the PDF and html outputs. Is there a way to do this from the Markdown document (YAML header)?

I've tried: 
header-includes:
  - \usepackage{fontspec}
  - \renewcommand{\familydefault}{\rmdefault}
  - \setmainfont{Pacetta Pro}

but I get an error when trying to load the "fontspec" package. I get the following error:

"Fatal fontspec error: "cannot-use-pdftex"
! 
! The fontspec package requires either XeTeX or LuaTeX."

Any ideas?
Thank you!

**Comment by eddelbuettel** [comment]
That is a feature.

**Comment by eddelbuettel** [comment]
As for 'requires XeTeX or LuaTeX':  sounds like your error.  That _can_ be specifiied in the yaml, ie all my beamer presentations (to pdf) use this:

```
    latex_engine: xelatex
```

which you can set within the `output:` section.

**Comment by alexsolisc** [comment]
Meaning... it can't be changed, right? Too bad. As for the 'requires XeTeX or LuaTeX', I did include 
`latex_engine: xelatex`
in my YAML header, but the error persists.

Many thanks Mr. Eddelbuettel!

**Comment by alexsolisc** [comment]
One last question: can you arrange for figures to span across the entire page by using the `fig.fullwidth` chunk option as in the Tufte templates?

**Comment by eddelbuettel** [comment]
Well the example that ships with `tint` has a full-width figure.

**Comment by alexsolisc** [comment]
Right! Just saw it, sorry for the dumb question. Thanks for all the help!

Best,
Alexis

**Comment by eddelbuettel** [comment]
Ok, I just tried.  The package has `pdflatex` hard-wired. You cannot override in the header.

And as I somewhat flippantly said above, that is indeed ... the way it is hardwired as this style in somewhat opionated in its look and feel.  So I think your best here is to ... derived your own package based on tint making your changes.  It's not that hard.

**Comment by eddelbuettel** [comment]
Or, for arguments sake, just do it by hand.  On my box this command is running:

```
/usr/lib/rstudio-server/bin/pandoc/pandoc +RTS -K512m -RTS tintEx.utf8.md --to latex \
   --from markdown+autolink_bare_uris+ascii_identifiers+tex_math_single_backslash \
   --output tintEx.tex --template \
          /usr/local/lib/R/sitel-ibrary/tint/rmarkdown/templates/tintPdf/resources/tintPdf-template.tex \
  --highlight-style tango --pdf-engine pdflatex --natbib --variable 'documentclass:tufte-handout' 
```

You could edit that by hand and try as a first test...

**Comment by alexsolisc** [comment]
I see! Ok, I'll try to do that. By the way, congratulations on the excellent `tint` package, I love it! The only change I would make is to include font customization. I'll try to edit the code you just posted.

Thanks!

**Comment by eddelbuettel** [comment]
Another way to experiment for a bit may be to set `keep_tex: true` and then just edit the .tex file for a bit.

**Comment by alexsolisc** [comment]
This worked! I just keep the tex file and then change the latex engine and font from there. Thank you!

**Comment by erex** [comment]
#### Secrets of `tint` and fonts for Windows users

If using `tint` on a Windows platform with Miktex 2.9, the RobotoCondensed font set needs to be downloaded from [Google](https://fonts.google.com/specimen/Roboto+Condensed), but that is not sufficient.

A final step needs to be performed after installing the fonts as described on [Stackoverflow](https://tex.stackexchange.com/questions/404085/installing-a-new-font-for-miktex-2-9-texmaker-source-sans-pro) 

> Call on a command line updmap. – Ulrike Fischer Dec 1 '17 at 17:37

With the fonts installed and registered, `tint` can make lovely Rmarkdown pdfs.

**Comment by eddelbuettel** [comment]
Sure. After download you also need to update the index files to communicate to your tex installation that you actually added a font.

I believe TeXLive and also MikTeX have GUI frontend tools to manage this; presumably they download _and_ index.  I also believe that the `tinytex` package aims to help here.   I cannot help adding more specific documentation as I do _not_ work on Windows.

Personally speaking I find this all a little opaque, but to complete this, when I package fonts for my Debian / Ubuntu system the same two step happens: download (which I take care of) and update font lists (for which the packaging has a hook into, on those systems, into calling `fc-cache -f` to force a recomputation of the font cache index ).

In short, these are theme packages for intermediate users who can install additional fonts.
